### PR TITLE
fix: Updates messaging on disable expiry for learner credit

### PR DESCRIPTION
**Pull request by brobro10000**
Updates the language of the helptext for the `disable_expiry_messaging_for_learner_credit` field on the enterprise customer model to accurately convey the behavior of the toggle.

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/openedx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/openedx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/openedx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - Trigger the '[Upgrade one Python dependency](https://github.com/openedx/edx-platform/actions/workflows/upgrade-one-python-dependency.yml)' action against master in edx-platform with new version number to generate version bump PR
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -17,6 +17,10 @@ Unreleased
 ----------
 * nothing unreleased
 
+[4.21.4]
+---------
+* feat: Updates messaging on disable expiry for learner credit
```
Comment (adamstankiewicz) [inline CHANGELOG.rst:22]: nit: if this a `feat`, then the minor version should be bumped, not the patch version (i.e., `4.22.0` vs. `4.21.4`). Given this is updating help text, seems like it'd be more in line with `fix` (i.e., not really introducing net-new functionality).

**Review by iloveagent57 — APPROVED** [review]

File: enterprise/models.py [hunk 2]
```diff
@@ -316,8 +316,9 @@ class Meta:
         verbose_name="Disable expiration messaging for learner credit",
         default=False,
         help_text=_("If checked, learners and admins will not receive notifications leading up to the expiration "
-                    "date for learner credit plans. Notifications signaling the expiration (and loss of usability) "
-                    "itself will still appear.")
+                    "date for learner credit plans. This includes removing the expiration date from the subsidy "
+                    "box for upcoming expiratons. The subsidy summary box will not display for expired plans. "
+                    "Other, notifications signaling the expiration (and loss of usability) itself will still appear.")
```
Comment (iloveagent57) [inline enterprise/models.py:321]: Nit: stray comma after "Other".  Whether you want to regen migrations to fix this is up to you :p

Comment (brobro10000) [reply enterprise/models.py]: Good catch. Regened 👍🏽

File: enterprise/migrations/0217_alter_enterprisecustomer_disable_expiry_messaging_for_learner_credit_and_more.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    dependencies = [
+        ('enterprise', '0216_enterprisegroupmembership_errored_at_and_more'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='enterprisecustomer',
+            name='disable_expiry_messaging_for_learner_credit',
+            field=models.BooleanField(default=False, help_text='If checked, learners and admins will not receive notifications leading up to the expiration date for learner credit plans. This includes removing the expiration date from the subsidy box for upcoming expiratons. The subsidy summary box will not display for expired plans. Other notifications signaling the expiration (and loss of usability) itself will still appear.', verbose_name='Disable expiration messaging for learner credit'),
```
Comment (adamstankiewicz) [inline enterprise/migrations/0217_alter_enterprisecustomer_disable_expiry_messaging_for_learner_credit_and_more.py:16]: nit: "expiratons" -> "expirations"

**Review by adamstankiewicz — APPROVED** [review]

**Pull request merged**